src/data/README.md

```diff
@@ -6,8 +6,100 @@ The `src/data/` directory contains all of the product data needed for various vi
 
 The `<product slug>.json` files (e.g. `boundary.json`, `consul.json`) contain metadata for each product that is used across multiple views, including their .io site and Dev Portal pages.
 
+### .io properties
+
 The canonical list of options (including a brief description of their usage) can be found in the [type definition for `ProductData`](./types.d.ts).
 
+### DevDot properties
```

Comment:
Since these props live in files described by `src/data/types.d.ts` would you mind adding these properties there as well (feel free to use simplified descriptions since they're really well documented here)? That way we can enforce type safety when accessing these DevDot properties.

Reply:
I'm not sure what you mean. Are you asking for duplicate types added to `src/data/types.d.ts`?

Also not sure what you mean by "That way we can enforce type safety when accessing these DevDot properties" because we do enforce type safety (just in a manual way) where we use `{product}.json` files in DevDot. One example here, where we cast the data imported as `Product`: https://github.com/hashicorp/dev-portal/blob/amb.add-dev-dot-json-data-docs/src/pages/waypoint/index.tsx.

For context there are plans to improve the `Product` type and break up the interface so it's clearer where the different properties are used, and hopefully that can happen next week in the frontend refinement cycle I'm in. 🤞🏻

Reply:
Ah okay! It looks like we have two duplicate definitions for the type of `src/data/<productSlug>.json`. I'm happy to follow up this PR adding the properties and docs from `src/data/types.d.ts` to `src/types/product.ts` and removing `src/data/types.d.ts` so we have a single source of truth for what those files look like.

Reply:
Question on that, does having `types.d.ts` in `src/data` get us type checking automatically? I think we need a mechanism that always checks that the required data is provided in these JSON files since they're easily changeable by anyone with write access.

Also, I like the idea of calling a single merged type `ProductData`. The struggle with what we currently call `Product` is that it had properties that don't really make sense with the name of `Product`. The type is used in a _ton_ of files though, so maybe the merge you're proposing is a good first step before renaming anything. `ProductData` has a much smaller footprint in the codebase. 😅

Reply:
It does not, however we _can_ add a JSON Schema file that'll get us fancy stuff in supported editors (like VS Code), so that could help with all of our various `.json` files we'll be allowing authors to edit. That's a separate file though, so even in that case we can get rid of `types.d.ts` (which, as you pointed out, is only being referenced in `lib/cms`, so it should be a really easy fix to point that to `types/product.ts` instead). Happy to explore these in a follow up PR, so this one has a 👍 from me!

Reply:
Sounds good! We can chat about this more next week. 😊